packages/aic/src/document.ts

```diff
@@ -0,0 +1,318 @@
+        "sig": "z4PWrudF88YEBqPQPhfWpiL7MFqhHhCWuKn8Ri4MyKZRcRLbSMFgP3WYKTBsfnq5TXGR9wbjJKKPSb7i1kQgisHzf"
+      },
+      "3j5r-ts5-ofsm-2c": {
+        "prev": "3j5r-ts5-ocuu-2c",
+        "patches": [ [ "del", [ "adx/account_keys" ] ] ],
+        "key": "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV",
+        "sig": "zSFwunTYyAh5ePkDgJ1Gw1nCtnqpVs1QzfXzkB9bHvoLoYKNeKsfeFHvToKT6tLiB5y9H5CBH1jgeYPtK57kodh7"
+      },
+      "3j5r-ts5-ojpm-2c": {
+        "prev": "3j5r-ts5-ofsm-2c",
+        "patches": [ [ "put", [ "hello" ], "world" ] ],
+        "key": "did:key:zDnaeycJUNQugcrag1WmLePtK9agLYLyXvscnQM4FHm1ASiRV",
+        "sig": "z431h9jjokS7TJ3VWT6XtbSUfTFVGUrLQi2uNMU4PbRXSNLoZS13dkkyeBrnfRrHdpiZhYGx6LCYSw5GgdEiDfEm9"
+      }
+    },
+    "key": "did:key:zDnaeeL44gSLMViH9khhTbngNd9r72MhUPo4WKPeSfB8xiDTh",
+    "sig": "z3naShLa1KV5ZV5raHbuAAC2zjjyBbTp4htmCA42GKVLDQi7kSnFCteg7n5ap5uJMpsjdxLgL8E6kRR9cqyt6zqK6"
+  }
+makes
+  {
+    'adx/recovery_keys': [ 'did:key:zDnaeYHbbWiaCwAXvHXRyVqENDv8Sr3fwHW8P5eakkz4MqsTa'],
+    id: 'did:aic:zrr5lxhs4rjlowv5',
+    name: 'aaron.blueskyweb.xyz'
+  }
+*/
+
+export const updateTick = async (
+  did: string, // the did:aic: being updated
+  tid: TidString, // the consenus time of the tick's generation
+  candidateDiff: Diff | Document | null, // null when only updating tid
+  stored_tick: Tick | null, // null when there is not db entry (did init)
+  key: Asymmetric, // the consortium passes in the key
+): Promise<Tick | ErrorMessage> => {
+  // @TODO validate that tid > all tids in the diffs
+  if (stored_tick === null) {
+    // this is the  initial registration of a did:aic
+    return registerNewDid(did, tid, candidateDiff, key)    
+  }
+
+  // since stored_tick is not null this is an update
+  // of a registered did:aic
+  if (!(await validateSig(stored_tick, key))) {
+    // this means the consortium data base has been corupted
+    return { error: 'bad signature: stored tick' }
+  }
+  if (stored_tick.key != key.did()) {
+    // the consortium should not update a tick signed by a difrent consortium
+    // this logic may get more complicated if the consortium is mid key rotation
+    return { error: 'mismatch: consortium key, stored tick key' }
+  }
+  if (stored_tick.did != did) {
+    // databace index is coruptend
+    return {
+      error: 'stored tick not for did',
+      did: did,
+      stored_tick_did: stored_tick.did,
+    }
+  }
+  // stored_tick is now validated by consortium key and url
+  return updateExistingValidatedDid(tid, candidateDiff, stored_tick, key)
+}
+
+const registerNewDid = async (
+  did: string,
+  tid: TidString,
+  candidateDiff: Document | null,
+  key: Asymmetric,
+): Promise<Tick | ErrorMessage> => {
+  if (candidateDiff === null) {
+    // since candidateDiff is null
+    // this is a request for a fresh tick of a non-exixtant did
+    // return a signed affirmation that the did dose not exist
+    return (await sign(
```

Comment:
I'm not sure we need to be signing error messages.

I understand the appeal for ticks. But for errors, isn't this handled by HTTPS?

Reply:
The real problem here is that it is not really an error.

It is an affirmative statement that the DID is not registered.
This is useful as a cacheable value that the DID is not registered but if the DID has never been updated in any way then it is still in the initial state.
That means that it can be validated by just using the hash of the initial state.

If I want to keep my ID private until I need to do the first update then the only info a validator needs to know that they got the current version of the doc it that there are no updates which this message proofs.